lib/uring/uring.mli

```diff
@@ -93,40 +99,138 @@ module type FLAGS = sig
 
-(** Flags that can be passed to openat2. *)
+(** Flags that can be passed to {!openat2}. *)
 module Open_flags : sig
   include FLAGS
 
-  val empty : t
   val append : t
+  (** [append] repositions the file offset to the end of the file before
+      every write. This should be used with caution with io_uring.
+      @see <https://github.com/axboe/liburing/issues/32#issuecomment-1313220682> GitHub axboe/liburing#32 *)
+
   val cloexec : t
+  (** [cloexec] enables the close-on-exec flag for the new fd. *)
+
   val creat : t
+  (** [create] implies that if the pathname does not exist, it is
+      created as a regular file. *)
+
   val direct : t
+  (** [direct] disables the kernel buffer cache and performs IO
+      directly to and from the userspace buffers. *)
+
   val directory : t
+  (** [directory] causes the open operation to fail if the target
+      is not a directory. *)
+
   val dsync : t
+  (** [dsync] ensures that write operations on the file complete
+      according to the requirements of synchronised IO data integrity
+      completion. *)
+
   val excl : t
-  val largefile : t
+  (** [excl] is used alongside {!creat} to ensure that the file
+      is created as a result of the {!openat2} call, and otherwise
+      fails with a {!Unix.EEXIST} exception. The only exception where
+      [excl] can be used without {!creat} is when attempting to open
+      block devices. If the block device is otherwise mounted, then
+      the open will fail with {!Unix.EBUSY}. *)
+
   val noatime : t
+  (** [noatime] signals that the file access time should not be updated
+      when the file is read from. See {!Statx.atime_nsec}. *)
+
   val noctty : t
+  (** [noctty] ensures that if the path refers to a tty, it will not
+      be assigned as the controlling terminal even if one is not present. *)
+
   val nofollow : t
+  (** [nofollow] will cause the open to fail with {!Unix.ELOOP} if the
+      basename of the path is a symbolic link. *)
+
   val nonblock : t
+  (** [nonblock] will open the file in non-blocking mode. *)
+
   val path : t
+  (** [path] will obtain a fd that can only be used to either indicate
+      a location in a filesystem tree, or perform operations at the fd
+      level. The file is not opened, and so any IO operations on the file
+      will fail. [path] is only used with {!cloexec}, {!directory} and
+      {!nofollow}, and any other flags will be ignored. *)
+
   val sync : t
+  (** [sync] ensures that write operations on the file complete
+      according to the requirements of synchronised IO file integrity
+      completion. *)
+
   val tmpfile : t
+  (** [tmpfile] creates an anonymous temporary regular file. The pathname
+      must be a directory, within which an unnamed inode will be created.
+      If [tmpfile] is specified without {!excl}, then a subsequent
+      linkat call can move it permanently into the filesystem. *)
+
   val trunc : t
+  (** [trunc] will set the file size of 0 if the file already exists
```

Comment:
```suggestion
  (** [trunc] will set the file size to 0 if the file already exists
```

Reply:
fixed in f9769b8d1cfe14a619182467dec21aef1965b2f5